services/cd-service/pkg/service/tags.go

```diff
@@ -68,15 +68,12 @@ func (s *TagsServer) GetProductSummary(ctx context.Context, in *api.GetProductSu
 	for _, row := range summaryFromEnv {
 		for _, app := range response.Applications {
 			if row.App == app.Name {
-				length := len(app.Releases) - 1
-				if app.Releases[length].DisplayVersion != "" {
-					productVersion = append(productVersion, &api.ProductSummary{App: row.App, Version: app.Releases[length].DisplayVersion})
-				} else if app.Releases[length].SourceCommitId != "" {
-					productVersion = append(productVersion, &api.ProductSummary{App: row.App, Version: app.Releases[length].SourceCommitId})
-				} else {
-					productVersion = append(productVersion, &api.ProductSummary{App: row.App, Version: row.Version})
+				for _, release := range app.Releases {
+					if strconv.Itoa(int(release.Version)) == row.Version {
```

Comment:
this typecast should be avoided because we are loosing information. See the implementation of strconv.Itoa. 
```suggestion
					if strconv.FormatUint(release.Version, 10) == row.Version {
```

Reply:
(not crucial because it's unlike that we have more than ~2^32 releases, but should be fixed anyway)